From 3d90ab11a421b26d141d8833c59ab8af512ae226 Mon Sep 17 00:00:00 2001
From: Vihang Karajgaonkar <vihang@cloudera.com>
Date: Wed, 31 May 2017 15:58:33 -0700
Subject: [PATCH 1163/1363] CDH-54539 : Hive first run failing due to
 HiveCreateMetastoreTables task failure

Temporary fix for post-commit failure for 5.13.0

Change-Id: Iadd48376b2da284f452fa2e0e1c8482947714782
---
 .../scripts/upgrade/derby/cdh.upgrade.order.derby  |    3 ++-
 .../upgrade/derby/hive-schema-1.1.0.derby.sql      |    4 ++--
 ...de-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.derby.sql |    3 +++
 .../scripts/upgrade/mssql/cdh.upgrade.order.mssql  |    1 +
 .../upgrade/mssql/hive-schema-1.1.0.mssql.sql      |    4 ++--
 ...de-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.mssql.sql |    4 ++++
 .../scripts/upgrade/mysql/cdh.upgrade.order.mysql  |    1 +
 .../upgrade/mysql/hive-schema-1.1.0.mysql.sql      |    4 ++--
 ...de-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.mysql.sql |    4 ++++
 .../upgrade/oracle/cdh.upgrade.order.oracle        |    1 +
 .../upgrade/oracle/hive-schema-1.1.0.oracle.sql    |    4 ++--
 ...e-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.oracle.sql |    3 +++
 .../upgrade/postgres/cdh.upgrade.order.postgres    |    1 +
 .../postgres/hive-schema-1.1.0.postgres.sql        |    4 ++--
 ...1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.postgres.sql |    4 ++++
 15 files changed, 34 insertions(+), 11 deletions(-)
 create mode 100644 metastore/scripts/upgrade/derby/upgrade-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.derby.sql
 create mode 100644 metastore/scripts/upgrade/mssql/upgrade-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.mssql.sql
 create mode 100644 metastore/scripts/upgrade/mysql/upgrade-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.mysql.sql
 create mode 100644 metastore/scripts/upgrade/oracle/upgrade-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.oracle.sql
 create mode 100644 metastore/scripts/upgrade/postgres/upgrade-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.postgres.sql

diff --git a/metastore/scripts/upgrade/derby/cdh.upgrade.order.derby b/metastore/scripts/upgrade/derby/cdh.upgrade.order.derby
index 9c28af4..5c47368 100644
--- a/metastore/scripts/upgrade/derby/cdh.upgrade.order.derby
+++ b/metastore/scripts/upgrade/derby/cdh.upgrade.order.derby
@@ -1 +1,2 @@
-1.1.0-to-1.1.0-cdh5.12.0
\ No newline at end of file
+1.1.0-to-1.1.0-cdh5.12.0
+1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0
diff --git a/metastore/scripts/upgrade/derby/hive-schema-1.1.0.derby.sql b/metastore/scripts/upgrade/derby/hive-schema-1.1.0.derby.sql
index 484bf55..1fd8691 100644
--- a/metastore/scripts/upgrade/derby/hive-schema-1.1.0.derby.sql
+++ b/metastore/scripts/upgrade/derby/hive-schema-1.1.0.derby.sql
@@ -336,7 +336,7 @@ RUN '041-HIVE-16556.derby.sql';
 
 ALTER TABLE "APP".VERSION ADD COLUMN SCHEMA_VERSION_V2 VARCHAR(255);
 -- -------------------------------------------------------------------------------------------------------------------------------------------------
--- Record schema version. Should be the last step in the init script. Starting CDH-5.12.0 CDH version also needs to be recorded in the version table
+-- Record schema version. Should be the last step in the init script. Starting CDH-5.13.0 CDH version also needs to be recorded in the version table
 -- -------------------------------------------------------------------------------------------------------------------------------------------------
-INSERT INTO "APP"."VERSION" (VER_ID, SCHEMA_VERSION, SCHEMA_VERSION_V2, VERSION_COMMENT) VALUES (1, '1.1.0', '1.1.0-cdh5.12.0', 'Hive release version 1.1.0-cdh5.12.0');
+INSERT INTO "APP"."VERSION" (VER_ID, SCHEMA_VERSION, SCHEMA_VERSION_V2, VERSION_COMMENT) VALUES (1, '1.1.0', '1.1.0-cdh5.13.0', 'Hive release version 1.1.0-cdh5.13.0');
 
diff --git a/metastore/scripts/upgrade/derby/upgrade-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.derby.sql b/metastore/scripts/upgrade/derby/upgrade-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.derby.sql
new file mode 100644
index 0000000..b8dd29c
--- /dev/null
+++ b/metastore/scripts/upgrade/derby/upgrade-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.derby.sql
@@ -0,0 +1,3 @@
+-- Upgrade MetaStore schema from 1.1.0-cdh5.12.0 to 1.1.0-cdh5.13.0
+
+UPDATE "APP".VERSION SET SCHEMA_VERSION='1.1.0', SCHEMA_VERSION_V2='1.1.0-cdh5.13.0', VERSION_COMMENT='Hive release version 1.1.0-cdh5.13.0' where VER_ID=1;
diff --git a/metastore/scripts/upgrade/mssql/cdh.upgrade.order.mssql b/metastore/scripts/upgrade/mssql/cdh.upgrade.order.mssql
index d1fa6d5..5c47368 100644
--- a/metastore/scripts/upgrade/mssql/cdh.upgrade.order.mssql
+++ b/metastore/scripts/upgrade/mssql/cdh.upgrade.order.mssql
@@ -1 +1,2 @@
 1.1.0-to-1.1.0-cdh5.12.0
+1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0
diff --git a/metastore/scripts/upgrade/mssql/hive-schema-1.1.0.mssql.sql b/metastore/scripts/upgrade/mssql/hive-schema-1.1.0.mssql.sql
index a47c0b9..2c867e8 100644
--- a/metastore/scripts/upgrade/mssql/hive-schema-1.1.0.mssql.sql
+++ b/metastore/scripts/upgrade/mssql/hive-schema-1.1.0.mssql.sql
@@ -946,6 +946,6 @@ ALTER TABLE TXN_COMPONENTS  WITH CHECK ADD FOREIGN KEY(TC_TXNID) REFERENCES TXNS
 
 ALTER TABLE VERSION ADD SCHEMA_VERSION_V2 VARCHAR(255);
 -- -----------------------------------------------------------------------------------------------------------------------------------------
--- Record schema version. Should be the last step in the init script. Starting CDH-5.12.0 CDH version must be included in the schema version
+-- Record schema version. Should be the last step in the init script. Starting CDH-5.13.0 CDH version must be included in the schema version
 -- -----------------------------------------------------------------------------------------------------------------------------------------
-INSERT INTO VERSION (VER_ID, SCHEMA_VERSION, SCHEMA_VERSION_V2, VERSION_COMMENT) VALUES (1, '1.1.0', '1.1.0-cdh5.12.0', 'Hive release version 1.1.0-cdh5.12.0');
+INSERT INTO VERSION (VER_ID, SCHEMA_VERSION, SCHEMA_VERSION_V2, VERSION_COMMENT) VALUES (1, '1.1.0', '1.1.0-cdh5.13.0', 'Hive release version 1.1.0-cdh5.13.0');
diff --git a/metastore/scripts/upgrade/mssql/upgrade-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.mssql.sql b/metastore/scripts/upgrade/mssql/upgrade-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.mssql.sql
new file mode 100644
index 0000000..2c15a6c
--- /dev/null
+++ b/metastore/scripts/upgrade/mssql/upgrade-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.mssql.sql
@@ -0,0 +1,4 @@
+SELECT 'Upgrading MetaStore schema from 1.1.0-cdh5.12.0 to 1.1.0-cdh5.13.0' AS MESSAGE;
+
+UPDATE VERSION SET SCHEMA_VERSION='1.1.0', SCHEMA_VERSION_V2='1.1.0-cdh5.13.0', VERSION_COMMENT='Hive release version 1.1.0-cdh5.13.0' where VER_ID=1;
+SELECT 'Finished upgrading MetaStore schema from 1.1.0-cdh5.12.0 to 1.1.0-cdh5.13.0' AS MESSAGE;
diff --git a/metastore/scripts/upgrade/mysql/cdh.upgrade.order.mysql b/metastore/scripts/upgrade/mysql/cdh.upgrade.order.mysql
index d1fa6d5..5c47368 100644
--- a/metastore/scripts/upgrade/mysql/cdh.upgrade.order.mysql
+++ b/metastore/scripts/upgrade/mysql/cdh.upgrade.order.mysql
@@ -1 +1,2 @@
 1.1.0-to-1.1.0-cdh5.12.0
+1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0
diff --git a/metastore/scripts/upgrade/mysql/hive-schema-1.1.0.mysql.sql b/metastore/scripts/upgrade/mysql/hive-schema-1.1.0.mysql.sql
index f54fea94..2081978 100644
--- a/metastore/scripts/upgrade/mysql/hive-schema-1.1.0.mysql.sql
+++ b/metastore/scripts/upgrade/mysql/hive-schema-1.1.0.mysql.sql
@@ -828,9 +828,9 @@ SOURCE 041-HIVE-16556.mysql.sql;
 
 ALTER TABLE VERSION ADD COLUMN SCHEMA_VERSION_V2 VARCHAR(255);
 -- ----------------------------------------------------------------------------------------------------------------------------------------
--- Record schema version. Should be the last step in the init script. Starting CDH-5.12.0 the schema version should also include CDH version
+-- Record schema version. Should be the last step in the init script. Starting CDH-5.13.0 the schema version should also include CDH version
 -- -----------------------------------------------------------------------------------------------------------------------------------------
-INSERT INTO VERSION (VER_ID, SCHEMA_VERSION, SCHEMA_VERSION_V2, VERSION_COMMENT) VALUES (1, '1.1.0', '1.1.0-cdh5.12.0', 'Hive release version 1.1.0-cdh5.12.0');
+INSERT INTO VERSION (VER_ID, SCHEMA_VERSION, SCHEMA_VERSION_V2, VERSION_COMMENT) VALUES (1, '1.1.0', '1.1.0-cdh5.13.0', 'Hive release version 1.1.0-cdh5.13.0');
 
 /*!40101 SET character_set_client = @saved_cs_client */;
 /*!40103 SET TIME_ZONE=@OLD_TIME_ZONE */;
diff --git a/metastore/scripts/upgrade/mysql/upgrade-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.mysql.sql b/metastore/scripts/upgrade/mysql/upgrade-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.mysql.sql
new file mode 100644
index 0000000..7de6fd0
--- /dev/null
+++ b/metastore/scripts/upgrade/mysql/upgrade-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.mysql.sql
@@ -0,0 +1,4 @@
+-- Upgrade MetaStore schema from 1.1.0-cdh5.12.0 to 1.1.0-cdh5.13.0
+
+UPDATE VERSION SET SCHEMA_VERSION='1.1.0', SCHEMA_VERSION_V2='1.1.0-cdh5.13.0', VERSION_COMMENT='Hive release version 1.1.0-cdh5.13.0' where VER_ID=1;
+SELECT 'Finished upgrading MetaStore schema from 1.1.0-cdh5.12.0 to 1.1.0-cdh5.13.0' AS ' ';
diff --git a/metastore/scripts/upgrade/oracle/cdh.upgrade.order.oracle b/metastore/scripts/upgrade/oracle/cdh.upgrade.order.oracle
index d1fa6d5..5c47368 100644
--- a/metastore/scripts/upgrade/oracle/cdh.upgrade.order.oracle
+++ b/metastore/scripts/upgrade/oracle/cdh.upgrade.order.oracle
@@ -1 +1,2 @@
 1.1.0-to-1.1.0-cdh5.12.0
+1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0
diff --git a/metastore/scripts/upgrade/oracle/hive-schema-1.1.0.oracle.sql b/metastore/scripts/upgrade/oracle/hive-schema-1.1.0.oracle.sql
index 2475fd8..0963ab2 100644
--- a/metastore/scripts/upgrade/oracle/hive-schema-1.1.0.oracle.sql
+++ b/metastore/scripts/upgrade/oracle/hive-schema-1.1.0.oracle.sql
@@ -789,6 +789,6 @@ CREATE INDEX FUNC_RU_N49 ON FUNC_RU (FUNC_ID);
 
 ALTER TABLE VERSION ADD SCHEMA_VERSION_V2 VARCHAR2(255);
 -- -----------------------------------------------------------------
--- Record schema version. Should be the last step in the init script. Starting CDH5.12.0 schema version must include the CDH version as well
+-- Record schema version. Should be the last step in the init script. Starting CDH5.13.0 schema version must include the CDH version as well
 -- -----------------------------------------------------------------
-INSERT INTO VERSION (VER_ID, SCHEMA_VERSION, SCHEMA_VERSION_V2, VERSION_COMMENT) VALUES (1, '1.1.0', '1.1.0-cdh5.12.0', 'Hive release version 1.1.0-cdh5.12.0');
+INSERT INTO VERSION (VER_ID, SCHEMA_VERSION, SCHEMA_VERSION_V2, VERSION_COMMENT) VALUES (1, '1.1.0', '1.1.0-cdh5.13.0', 'Hive release version 1.1.0-cdh5.13.0');
diff --git a/metastore/scripts/upgrade/oracle/upgrade-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.oracle.sql b/metastore/scripts/upgrade/oracle/upgrade-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.oracle.sql
new file mode 100644
index 0000000..93a5f52
--- /dev/null
+++ b/metastore/scripts/upgrade/oracle/upgrade-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.oracle.sql
@@ -0,0 +1,3 @@
+-- Upgrade MetaStore schema from 1.1.0-cdh5.12.0 to 1.1.0-cdh5.13.0
+UPDATE VERSION SET SCHEMA_VERSION='1.1.0', SCHEMA_VERSION_V2='1.1.0-cdh5.13.0', VERSION_COMMENT='Hive release version 1.1.0-cdh5.13.0' where VER_ID=1;
+SELECT 'Finished upgrading MetaStore schema from 1.1.0-cdh5.12.0 to 1.1.0-cdh5.13.0' AS Status from dual;
diff --git a/metastore/scripts/upgrade/postgres/cdh.upgrade.order.postgres b/metastore/scripts/upgrade/postgres/cdh.upgrade.order.postgres
index d1fa6d5..5c47368 100644
--- a/metastore/scripts/upgrade/postgres/cdh.upgrade.order.postgres
+++ b/metastore/scripts/upgrade/postgres/cdh.upgrade.order.postgres
@@ -1 +1,2 @@
 1.1.0-to-1.1.0-cdh5.12.0
+1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0
diff --git a/metastore/scripts/upgrade/postgres/hive-schema-1.1.0.postgres.sql b/metastore/scripts/upgrade/postgres/hive-schema-1.1.0.postgres.sql
index c48c042..837c519 100644
--- a/metastore/scripts/upgrade/postgres/hive-schema-1.1.0.postgres.sql
+++ b/metastore/scripts/upgrade/postgres/hive-schema-1.1.0.postgres.sql
@@ -1494,6 +1494,6 @@ GRANT ALL ON SCHEMA public TO PUBLIC;
 
 ALTER TABLE "VERSION" ADD COLUMN "SCHEMA_VERSION_V2" VARCHAR(255);
 -- ------------------------------------------------------------------------------------------------------------------------------
--- Record schema version. Should be the last step in the init script. Starting CDH-5.12.0 schema version must include CDH version
+-- Record schema version. Should be the last step in the init script. Starting CDH-5.13.0 schema version must include CDH version
 -- ------------------------------------------------------------------------------------------------------------------------------
-INSERT INTO "VERSION" ("VER_ID", "SCHEMA_VERSION", "SCHEMA_VERSION_V2", "VERSION_COMMENT") VALUES (1, '1.1.0', '1.1.0-cdh5.12.0', 'Hive release version 1.1.0-cdh5.12.0');
+INSERT INTO "VERSION" ("VER_ID", "SCHEMA_VERSION", "SCHEMA_VERSION_V2", "VERSION_COMMENT") VALUES (1, '1.1.0', '1.1.0-cdh5.13.0', 'Hive release version 1.1.0-cdh5.13.0');
diff --git a/metastore/scripts/upgrade/postgres/upgrade-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.postgres.sql b/metastore/scripts/upgrade/postgres/upgrade-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.postgres.sql
new file mode 100644
index 0000000..2629eee
--- /dev/null
+++ b/metastore/scripts/upgrade/postgres/upgrade-1.1.0-cdh5.12.0-to-1.1.0-cdh5.13.0.postgres.sql
@@ -0,0 +1,4 @@
+SELECT 'Upgrading MetaStore schema from 1.1.0-cdh5.12.0 to 1.1.0-cdh5.13.0';
+
+UPDATE "VERSION" SET "SCHEMA_VERSION"='1.1.0', "SCHEMA_VERSION_V2"='1.1.0-cdh5.13.0', "VERSION_COMMENT"='Hive release version 1.1.0-cdh5.13.0' where "VER_ID"=1;
+SELECT 'Finished upgrading MetaStore schema from 1.1.0-cdh5.12.0 to 1.1.0-cdh5.13.0';
-- 
1.7.9.5

